src/main/java/org/mockito/internal/session/DefaultMockitoSessionBuilder.java

```diff
@@ -27,11 +55,27 @@ public MockitoSessionBuilder strictness(Strictness strictness) {
         return this;
     }
 
+    @Override
+    public MockitoSessionBuilder logger(MockitoSessionLogger logger) {
+        this.logger = logger;
+        return this;
+    }
+
     @Override
     public MockitoSession startMocking() {
         //Configure default values
-        Object effectiveTest = this.testClassInstance == null ? new Object() : this.testClassInstance;
+        List<Object> effectiveTestClassInstances;
+        String effectiveName;
+        if (testClassInstances.isEmpty()) {
+            effectiveTestClassInstances = emptyList();
+            effectiveName = this.name == null ? UUID.randomUUID().toString() : this.name;
```

Comment:
Random String may not provide great user experience. Can we make it more self-describing in case our users see that name? Can we just have empty String or "test". With the current functionality of Mockito, are there any circumstances where random UUID will be visible to the user?

Reply:
It is visible in the emitted hints, e.g. in https://github.com/mockito/mockito/blob/c8f07c70a2e2be4714b4fc6a1a7f6e8b473b3bac/src/main/java/org/mockito/internal/junit/StubbingHint.java#L14-L17

How about using sth. like `"<unnamed session>"`?